plugins/tiddlywiki/tiddlyweb/save-wiki-button.tid

```diff
@@ -3,52 +3,47 @@ tags: $:/tags/PageControls
-	<span class="tc-dirty-indicator">
-		<$list filter="[<tv-config-toolbar-icons>match[yes]]">
-			<span class="tc-dirty-control tc-hide-when-dirty">
-				{{$:/core/images/cloud}}
-			</span>
-			<span class="tc-dirty-control tc-show-when-dirty">
-				{{$:/core/images/cloud-sync}}
-			</span>
-		</$list>
-		<$list filter="[<tv-config-toolbar-text>match[yes]]">
-			<span class="tc-btn-text"><$text text="Server status"/></span>
-		</$list>
-	</span>
+<span class="tc-dirty-indicator">
+<$list filter="[<tv-config-toolbar-icons>match[yes]]">
+{{$:/core/images/cloud}}
+</$list>
+<$list filter="[<tv-config-toolbar-text>match[yes]]">
+<span class="tc-btn-text"><$text text="Server status"/></span>
+</$list>
+</span>
 </$button>
 <$reveal state=<<qualify "$:/state/popup/save-wiki">> type="popup" position="below" animate="yes">
-	<div class="tc-drop-down">
-		<$reveal state="$:/status/IsLoggedIn" type="nomatch" text="yes">
-			<$button message="tm-login" class="tc-btn-invisible">
-				{{$:/core/images/unlocked-padlock}}<$text text=" "/>Login
-			</$button>
-		</$reveal>
-		<$reveal state="$:/status/IsLoggedIn" type="match" text="yes">
-			<p>
-				Currently logged in<$reveal state="$:/status/UserName" type="nomatch" text="" default=""> as <$text text={{$:/status/UserName}}/></$reveal>
-			</p>
-			<$button tooltip="Get latest changes from the server" aria-label="Refresh from server" class="tc-btn-invisible">
-				<$action-sendmessage $message="tm-server-refresh"/>
-				{{$:/core/images/refresh-button}}<$text text=" "/><span clas]s="tc-btn-text"><$text text="Refresh from server"/></span>
-			</$button>
-			<$button class="tc-btn-invisible">
-				<$wikify name="site-title" text={{$:/config/SaveWikiButton/Filename}}>
-					<$action-sendmessage $message="tm-download-file" $param={{$:/config/SaveWikiButton/Template}} filename=<<site-title>>/>
-				</$wikify>
-				{{$:/core/images/download-button}}<$text text=" "/>Save snapshot for offline use
-			</$button>
-			<$button message="tm-logout" class="tc-btn-invisible">
-				{{$:/core/images/cancel-button}}<$text text=" "/>Logout
-			</$button>
-		</$reveal>
-		<$button message="tm-copy-syncer-logs-to-clipboard" class="tc-btn-invisible">
-			{{$:/core/images/copy-clipboard}}<$text text=" "/>Copy syncer logs to clipboard
-		</$button>
-	</div>
+<div class="tc-drop-down">
+<$reveal state="$:/status/IsLoggedIn" type="nomatch" text="yes">
+<$button message="tm-login" class="tc-btn-invisible">
+{{$:/core/images/unlocked-padlock}} Login
+</$button>
+</$reveal>
+<$reveal state="$:/status/IsLoggedIn" type="match" text="yes">
+<p>
+Currently logged in<$reveal state="$:/status/UserName" type="nomatch" text="" default=""> as <$text text={{$:/status/UserName}}/></$reveal><$reveal state="$:/status/IsReadOnly" type="match" text="yes" default="no"> (read-only)</$reveal>
```

Comment:
If you had `\whitespace trim` problems here, ... that's OK. This text needs to be made translateable in the future. 

There should be _no `&nbsp;` and such stuff. Elements, that need spacing need a class + proper margin or padding. This makes the content "themeable" We have a some problems because of "hacky" UI elements. 

IMO `\whitespace trim` just points out the problems, that should be fixed.

Reply:
> Elements, that need spacing need a class + proper margin or padding

Yes, quite. But sadly I think we're stuck with using space characters for spacing in lots of situations for backwards compatibility.